infrastructure/zk/src/format_sql.ts

```diff
@@ -0,0 +1,161 @@
+    }
+    return formattedLines.join('\n');
+}
+
+function addIndent(query: string, indent: number) {
+    return query
+        .split('\n')
+        .map((line) => ' '.repeat(indent) + line)
+        .join('\n');
+}
+
+function formatRustStringQuery(query: string) {
+    const baseIndent = query.search(/\S/);
+    let endedWithAComma = query.trim().endsWith(',');
+
+    const rawQuery = extractQueryFromRustString(query);
+    const formattedQuery = formatQuery(rawQuery);
+    const reconstructedRustString = embedTextInsideRustString(formattedQuery);
+
+    return addIndent(reconstructedRustString, baseIndent) + (endedWithAComma ? ',' : '') + '\n';
+}
+
+async function formatFile(filePath: string, check: boolean) {
+    let content = await fs.promises.readFile(filePath, { encoding: 'utf-8' });
+    let linesToQuery = null;
+    let isInsideQuery = false;
+    let isRawString = false;
+    let builtQuery = '';
+
+    let modifiedFile = '';
+    for (const line of content.split('\n')) {
+        if (line.endsWith('sqlx::query!(')) {
+            linesToQuery = 1;
+            isRawString = false;
+            builtQuery = '';
+        }
+        if (line.endsWith('sqlx::query_as!(')) {
+            linesToQuery = 2;
+            isRawString = false;
+            builtQuery = '';
+        }
+
+        if (linesToQuery !== null) {
+            if (linesToQuery == 0) {
+                isInsideQuery = true;
+                linesToQuery = null;
+                if (line.includes('r#"')) {
+                    isRawString = true;
+                }
+            } else {
+                linesToQuery -= 1;
+            }
+        }
+
+        if (isInsideQuery) {
+            const rawStringQueryEnded = (line.endsWith('"#,') || line.endsWith('"#')) && builtQuery;
+            const regularStringQueryEnded = (line.endsWith('",') || line.endsWith('"')) && builtQuery;
+            builtQuery += line + '\n';
+            const lineEndIsNotEscape = !line.endsWith('\\"') && !line.endsWith('\\",');
+            if (
+                (isRawString && rawStringQueryEnded) ||
+                (!isRawString && regularStringQueryEnded && lineEndIsNotEscape)
+            ) {
+                isInsideQuery = false;
+                let formattedQuery = formatRustStringQuery(builtQuery);
+                modifiedFile += formattedQuery;
+            }
+        } else {
+            modifiedFile += line + '\n';
+        }
+    }
+    modifiedFile = modifiedFile.slice(0, modifiedFile.length - 1);
+
+    if (!check) {
+        await fs.promises.writeFile(filePath, modifiedFile);
+    } else {
+        if (content !== modifiedFile) {
+            console.warn(`Sqlx query format issues found in ${filePath}.`);
+        }
+    }
+    return content === modifiedFile;
+}
+
+export async function formatSqlxQueries(check: boolean) {
+    process.chdir(`${process.env.ZKSYNC_HOME}`);
+    let { stdout: filesRaw } = await utils.exec('find core/lib/dal -type f -name "*.rs"');
```

Comment:
Nit: This may be OS-specific; it would be safer to use something like [`fs.readdir`](https://nodejs.org/dist/latest-v20.x/docs/api/fs.html#fspromisesreaddirpath-options). OTOH, we probably already have quite a lot *NIX-specific logic.